### PR TITLE
usm: http2: classification: Improve classification of gRPC

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

The PR improves the gRPC plaintext classification.
* The PR uses functions used for the decoding, to ensure we have a single version
* The PR extends the number of frames we filter from 10 to 90
* The PR simplifies the code to handle verification errors
* The PR supports more edge cases (literal headers without indexing or never indexed, dynamic table update)

### Motivation

Improved algorithm to classify the protocol.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/skb-common.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    // See: https://datatracker.ietf.org/doc/html/rfc7541#appendix-B for more details on huffman code.
+    *is_huffman_encoded = (current_char_as_number & 128) > 0;
+
+    return read_hpack_int_with_given_current_char(skb, skb_info, current_char_as_number, max_number_for_bits, out);
+}
+
+// Handles the case in which a header is not a pseudo header. We don't need to save it as interesting or modify our telemetry.
+// https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3
+// https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.4
+static __always_inline bool handle_non_pseudo_headers(struct __sk_buff *skb, skb_info_t *skb_info, __u64 index) {
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/skb-common.h:63]: Lets rename this function name to a better one

File: pkg/network/ebpf/c/protocols/grpc/helpers.h [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include "protocols/http2/skb-common.h"
 #include "protocols/grpc/defs.h"
 
-#define GRPC_MAX_FRAMES_TO_FILTER 10
+#define GRPC_MAX_FRAMES_TO_FILTER 90
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/grpc/helpers.h:12]: lets add documentation for both of the consts as we do in the decoding part.

**Review by amitslavin — CHANGES_REQUESTED** [review]

**Review by amitslavin — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/grpc/helpers.h (hunk 2 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/grpc/helpers.h:12]: done

**Review by amitslavin — APPROVED** [review]

**Pull request merged**